kiwi/storage/loop_device.py

```diff
@@ -98,12 +101,13 @@ def create(self, overwrite: bool = True):
         )
         self.node_name = loop_call.output.rstrip(os.linesep)
 
-    def __del__(self):
+    def __exit__(self, exc_type, exc_value, exc_traceback):
         if self.node_name:
-            log.info('Cleaning up %s instance', type(self).__name__)
             try:
                 Command.run(['losetup', '-d', self.node_name])
             except Exception:
                 log.warning(
-                    'loop device %s still busy', self.node_name
+                    f'loop device {self.node_name} still busy'
                 )
+            # exception got handled, return gracefully
+            return True
```

Comment:
This is dangerous, returning `True` from `__exit__()` indicates that the exception thrown in the `with` block got handled. I.e. you are hereby swallowing **all** exceptions raised in every `with LoopDevice() as loop_provider:` block. I am pretty certain we do **not** want that.

```suggestion
```

Reply:
yep makes a lot sense, I fixed it